### ISSUE TITLE
azurerm_resource_group_template_deployment ignoring parameters after logic app creation 

### ISSUE THREAD

**Issue by kjakah** [terraform-providers/terraform-provider-azurerm#9840]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform: 0.14.2
AzureRM: 2.40.0

### Affected Resource(s)

* `azurerm_resource_group_template_deployment`
* `azurerm_logic_app_workflow`

### Terraform Configuration Files

```hcl
#versions.tf
terraform {
  required_version = "~> 0.14.2"

  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 2.40.0"
    }
  }
}

provider azurerm {
  features {}
  skip_provider_registration = true
}

#variables.tf
variable "location" {
  type    = string
  default = "eastus 2"
}


variable "prefix" {
  type    = string
  default = "kjdemo"
}



#main.tf
# Create a resource groups
resource "azurerm_resource_group" "rg" {
  name     = var.prefix
  location = var.location
}


# Logic app create:
resource "azurerm_logic_app_workflow" "logicappaas" {
  name                = "logic-${var.prefix}"
  location            = var.location
  resource_group_name = azurerm_resource_group.rg.name
}

#************************************************
#**NOTE: 
[createLogicAppsTEST.zip](https://github.com/terraform-providers/terraform-provider-azurerm/files/5681722/createLogicAppsTEST.zip)

The createLogicAppsTEST.json file is attached
#************************************************

# Then detailed ARM deployment of the same name parameter:
resource "azurerm_resource_group_template_deployment" "templateTEST" {
  name                = "arm-Deployment"
  resource_group_name = azurerm_resource_group.rg.name
  depends_on          = [azurerm_logic_app_workflow.logicappaas]
  deployment_mode     = "Incremental" 
  template_content    = file("${path.module}/createLogicAppsTEST.json")
  parameters_content = jsonencode({ 
    logic_app_name = { value = "logic-${var.prefix}" }
  })

}

```

### Expected Behaviour

The resource should be created on the first run given parameters passed from azurerm_resource_group_template_deployment to the ARM template. Then each subsequent run with no changes should yield no change to the resource and should not error out.

### Actual Behaviour

The Logic Apps resource is created just fine on the first run, no problem. After the first run, it appears terraform provider is ignoring the parameters passed from azurerm_resource_group_template_deployment and relies on the defaultValue property within the ARM template. If I remove the defaultvalue property from ARM template, it will create just fine on the first run, but on any run after the resource creation it will error out with the error seen in the attached png.
![Error](https://user-images.githubusercontent.com/48288701/101957991-7eb59d80-3bd0-11eb-815c-c3e9760d6af0.png)

In short Terraform appears to ignore the parameters I am passing and use defaultValue in the ARM template, on any run passed first resource creation, which makes a multi-tiered environment impossible to maintain given defaultvalue is hard-coded in the ARM template

### Steps to Reproduce

Run Terraform apply twice, once to create the resource, then to see the error:
1. `terraform apply`
2. `terraform apply`

**Comment by ruandersMSFT** [comment]
@kjakah The following should work utilizing templatefile:

```hcl
resource "azurerm_resource_group_template_deployment" "templateTEST" {
  name                = "arm-Deployment"
  resource_group_name = azurerm_resource_group.rg.name
  depends_on          = [azurerm_logic_app_workflow.logicappaas]
  deployment_mode     = "Incremental" 
  template_content    = file("${path.module}/createLogicAppsTEST.json")
  parameters_content = templatefile("${path.module}/createLogicAppsTEST-Parameters.json",
    {
      logic_app_name = "logic-${var.prefix}" 
    }
  )
}
```

Where the contents of the createLogicAppsTEST-Parameters.json file residing in the module path contains:

```hcl
{
  "logic_app_name": {
      "value": "${logic_app_name}"
  }
}
```

**Comment by kjakah** [comment]
> @kjakah The following should work utilizing the [templatefile](https://www.terraform.io/docs/configuration/functions/templatefile.html) function:
> 
> ```terraform
> resource "azurerm_resource_group_template_deployment" "templateTEST" {
>   name                = "arm-Deployment"
>   resource_group_name = azurerm_resource_group.rg.name
>   depends_on          = [azurerm_logic_app_workflow.logicappaas]
>   deployment_mode     = "Incremental" 
>   template_content    = file("${path.module}/createLogicAppsTEST.json")
>   parameters_content  = templatefile("${path.module}/createLogicAppsTEST-Parameters.json",
>     {
>       logic_app_name = "logic-${var.prefix}" 
>     }
>   )
> }
> ```
> 
> Where the contents of the createLogicAppsTEST-Parameters.json file residing in the module path contains:
> 
> ```terraform
> {
>   "logic_app_name": {
>       "value": "${logic_app_name}"
>   }
> }
> ```

Were you able to get this to run at least 2 times with my example given the error I posted above?  I received the exact same error as before

**Comment by kjakah** [comment]
I was able to get this to work as bleow, following the first example here: https://github.com/terraform-providers/terraform-provider-azurerm/issues/8756#issuecomment-704259517

resource block
```resource "azurerm_resource_group_template_deployment" "templateTEST" {
  name                = "arm-Deployment"
  resource_group_name = azurerm_resource_group.rg.name
  deployment_mode     = "Incremental"
  template_content = templatefile("${path.module}/arm/createLogicAppsTEST.json", {  # check within the 
    logicappname = "logic-${var.prefix}"
  })
}
```

createLogicAppsTEST.json file parameters location near top:
```
...
        "logic_app_name": {
            "type": "string",
            "defaultValue": "${logicappname}"
        }
...
```

Obvious problem is that we cannot source control the logic app now, given the need to go into the ARM template and manually fill in the defaultValue property with the proper variable, but at least we can pass variables to permit multi-tier deployments

**Comment by tombuildsstuff** [comment]
hi @kjakah 

Thanks for opening this issue :)

Taking a look through this appears to be a duplicate of #8840 - rather than having multiple issues open tracking the same thing I'm going to close this issue in favour of that one; would you mind subscribing to #8840 for updates?

Thanks!
